.gitlab-ci.yml

```diff
@@ -6,13 +7,75 @@ stages:
   - format
   - build
 
+.astyle:
+  cache:
+    paths:
+      - astyle_3.1_linux.tar.gz
+    when: always
+
+.show_hostname_and_groups: &show_hostname_and_groups
+  - export HOSTNAME=`hostname`
+  - echo -e "Hostname:" ${HOSTNAME}
+  - groups
+
+.download_astyle: &download_astyle
+  - wget --timestamping https://sourceforge.net/projects/astyle/files/astyle/astyle%203.1/astyle_3.1_linux.tar.gz
+
+.build_and_install_rankstr: &build_and_install_rankstr
+  # Build and install rankstr dependency
+  - git clone --config advice.detachedHead=false --depth 1 -- https://github.com/ECP-VeloC/rankstr.git
+  - cd rankstr
+  - mkdir build install && cd build
+  - cmake -DCMAKE_INSTALL_PREFIX=../install ..
+  - VERBOSE=1 make -j
+  - make install
+  - RANKSTR_INSTALL=${PWD}/../install
+  - cd ../../
+
+.script_lassen: &script_lassen
+  - *show_hostname_and_groups
+  - mkdir build-gitlab-lassen-"${CPUORGPU:?}" && mkdir install-gitlab-lassen-"${CPUORGPU:?}"
+  - cd build-gitlab-lassen-"${CPUORGPU:?}"
+  - echo -e "HOST_CONFIG:" ${HOST_CONFIG:?}
+  - echo -e "PWD:" ${PWD}
+  - cmake -C ${HOST_CONFIG} -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=../install-gitlab-lassen-"${CPUORGPU:?}" -DBUILD_TESTS=ON -DENABLE_WARNINGS=ON ../src
+  - make VERBOSE=1
+  - make install
+  - env CTEST_OUTPUT_ON_FAILURE=1 make test
+  - cd ..
+  - echo -e "PWD:" ${PWD}
+  - which cmake
+  - cmake --version
+  - cd install-gitlab-lassen-"${CPUORGPU:?}"/examples/using-with-cmake/c
+  - echo -e "PWD:" ${PWD}
+  - mkdir _test_build
+  - cd _test_build
+  - cmake ../
+  - make VERBOSE=1
+  - ./variorum-print-power-example
+  - cd ../../../using-with-make/c
+  - echo -e "PWD:" ${PWD}
+  - make VERBOSE=1
+  - ./variorum-print-power-example
+  - cd ../../using-with-make/cpp
+  - echo -e "PWD:" ${PWD}
+  - make VERBOSE=1
+  - ./variorum-print-power-example
+  - cd ../../using-with-make/fortran
+  - echo -e "PWD:" ${PWD}
+  - make VERBOSE=1
+  - ./ep.W.x
+
 build-quartz-format:
+ extends:
+   - .astyle
+ interruptible: true
```

Comment:
This makes GitLab cancel the job if a newer job (new commit on same branch) is scheduled.